cdap-master/src/test/java/io/cdap/cdap/master/environment/k8s/PreviewServiceMainTest.java

```diff
@@ -39,82 +42,223 @@
 import org.apache.twill.filesystem.LocalLocationFactory;
 import org.apache.twill.filesystem.Location;
 import org.apache.twill.filesystem.LocationFactory;
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.io.IOException;
 import java.io.InputStream;
+import java.lang.reflect.Type;
 import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
+import java.util.jar.Manifest;
 
 /**
  * Unit test for {@link AppFabricServiceMain}.
  */
 public class PreviewServiceMainTest extends MasterServiceMainTestBase {
   private static final Gson GSON = new Gson();
 
-  @Test
-  public void testPreviewService() throws Exception {
+  private static final Type ARTIFACT_SUMMARY_LIST = new TypeToken<List<ArtifactSummary>>() { }.getType();
+
+  @After
+  public void after() throws IOException {
+    deleteAllArtifact();
+  }
 
-    // Deploy the app artifact
+  @Test
+  public void testPreviewSimpleApp() throws Exception {
+    // Build the app
     LocationFactory locationFactory = new LocalLocationFactory(TEMP_FOLDER.newFolder());
     Location appJar = AppJarHelper.createDeploymentJar(locationFactory, PreviewTestApp.class);
 
+    // Deploy the app
     String artifactName = PreviewTestApp.class.getSimpleName();
     String artifactVersion = "1.0.0-SNAPSHOT";
+    deployArtifact(appJar, artifactName, artifactVersion);
 
-    HttpRequestConfig requestConfig = new HttpRequestConfig(0, 0, false);
-    URL url = getRouterBaseURI().resolve(String.format("/v3/namespaces/default/artifacts/%s", artifactName)).toURL();
-    HttpResponse response = HttpRequests.execute(
-      HttpRequest.post(url)
-        .withBody((ContentProvider<? extends InputStream>) appJar::getInputStream)
-        .addHeader("Artifact-Version", artifactVersion)
-        .build(), requestConfig);
-    Assert.assertEquals(HttpURLConnection.HTTP_OK, response.getResponseCode());
+    // Start the preview service main, which will use its own local datadir, thus should fetch artifact location
+    // from appFabric when running a preview
+    startService(PreviewServiceMain.class);
 
-    // have to stop AppFabric so that Preview can share the same leveldb table
```

Comment:
nice that we don't need this anymore

Reply:
Ideally we should have another test that spin up SQL. If time permits, will look into add that.